### ISSUE TITLE
Wigner D Matrix appears to be complex conjugated

### ISSUE THREAD

**Issue by ariaradick** [moble/SphericalFunctions.jl#42]
The easiest way to see this is to consider the expression

$$D^\ell_{m0}(\alpha,\beta,\gamma) = \sqrt{\frac{4 \pi}{2 \ell + 1}} Y_\ell^{m*}(\beta,\alpha)$$

if we calculate this for a given rotation, say $\alpha = \pi/3$ and $\beta = \pi/5$

```
using Quaternionic
using SphericalFunctions
θ = π/5
φ = π/3
q = from_euler_angles(φ, θ, 0.0)
[D_matrices(q,1)[WignerDindex(ell,m,0)] for ell in 0:1 for m in -ell:ell]
```
this returns
```
1.0 + 0.0im
  0.2078134688887268 - 0.3599434866124088im
  0.8090169943749475 + 0.0im
 -0.2078134688887268 - 0.3599434866124088im
```

If we use Mathematica to calculate the complex spherical harmonics on the RHS of this equation

```
Table[N[Sqrt[(4 \[Pi])/(2  ell + 1)]
    Conjugate[SphericalHarmonicY[ell, m, \[Pi]/5, \[Pi]/3]]], {ell, 0,
   1}, {m, -ell, ell}]
```
we get
```
1.
0.207813 + 0.359943 I
0.809017
-0.207813 + 0.359943 I
```

which is the complex conjugate of your D matrix.

**Comment by moble** [comment]
TL;DR: I disagree on the conventions.  I'll add more discussion of the conventions to the documentation on the next major release, but I'll probably just stick with the current choices.

---

Unfortunately, everything involving rotations is just a quagmire of poorly specified conventions, starting with the endless combinations of possible Euler angles, going through the meaning (and order) of the indices on the Wigner matrices, and even the meaning of the argument to the Wigner matrices.  So I'm not saying that my conventions are the best, but I do believe they are internally consistent.  And I've evidently chosen something different from what Mathematica has.

Probably the easiest way to compare conventions is to just look at the actual expressions in terms of Euler angles / spherical coordinates.

First, regarding the relationship between D and Y, my convention is given on the documentation page [here](https://moble.github.io/SphericalFunctions.jl/stable/sYlm/), where I don't have the conjugate that you have in your equation.  I chose my conventions for Y to be consistent with established formulas broadly used by the gravitational-wave community I work in.  FWIW, they're given [here](https://arxiv.org/abs/0709.0093).

Second, regarding D itself, I've added tests against Wigner's own expression as given in Eq. (15.27) of [the translation here](https://www.sciencedirect.com/bookseries/pure-and-applied-physics/vol/5/suppl/C).  His conventions for Euler angles agree with mine (despite the evidently swapped labels for $\alpha$ and $\gamma$ in Fig. 2), so we can directly compare results.  The only difference between our results is a factor of $(-1)^{m'-m}$, which is accounted for by the Condon-Shortley phase.  In particular, there is no complex conjugate between his version and mine.

Wigner relates D to Y below Eq. (19.8), where he does not include a conjugate.  Similarly, looking at Eqs. (15.8) and (A.11), we can see that he would have

$$Y\_{\ell,m}(\vartheta, \phi) = (\mathrm{const}) \cdot (-1)^m e^{i m \phi} \mathfrak{D}^{(\ell)}(\\{\phi, \vartheta, 0\\})\_{m,0}.$$

Again, that factor of $(-1)^m$ is a difference between his definitions and mine, but there's no conjugate.

Anyway, the point is that I'm not in the worst of company by choosing these conventions.  Of course, Mathematica and a few other more recent references do include the conjugate.  There's a pretty good argument to be made that they are making the more sensible choice, because one of the more important uses for $\mathfrak{D}$ is to rotate the $Y_{\ell,m}$, and Wigner's Eq. (A.8) looks like this:

$$Y\_{\ell,m}(\vartheta', \phi') = \mathfrak{D}^{(\ell)}(R)\_{m,k}^\ast Y_{\ell, k}(\vartheta, \phi).$$

That conjugate is pretty ugly, so I can certainly see the argument for flipping conventions.  At this point, I'm inclined to say that I've made my bed, so I'd better lie in it.  On the other hand, I am open to *persuasive* arguments to change in the next major release, though I will be skeptical.  In any case, I've now done a good bit of sweeping through references to understand conventions, so I'll be sure to include that in the docs for the next major release.
